File: 206Reverse Linked List.c
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */
struct ListNode* reverseList(struct ListNode* head) {
    if(head == NULL)
		return NULL;
    struct ListNode *prev = head;
	struct ListNode *pcur = head->next;
	while(pcur)
	{
		prev->next = pcur->next;
		pcur->next = head;
		head = pcur;
		pcur = prev->next;
	}
	return head;
}

File: source/206-Reverse Linked List/206Reverse Linked List.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution(object):
    def reverseList(self, head):
        """
        :type head: ListNode
        :rtype: ListNode
        """
        if None == head or None == head.next:
            return head
        prev = head
        pcur = head.next
        while(pcur):
            prev.next = pcur.next
            pcur.next = head
            head = pcur
            pcur = prev.next
        return head